JM/CF/Scripts/1_Core/CommunityFramework/Files/CF_File.c

```diff
@@ -0,0 +1,167 @@
+class CF_File : Managed
+{
+	protected string m_FileName;
+	protected string m_Folder;
+	protected bool m_IsDirectory;
+	protected bool m_IsHidden;
+	protected bool m_IsReadOnly;
+	protected bool m_IsValid;
+
+	/**
+	 * @param pattern The search pattern
+	 * @param [inout] files An array where the list of files will be written to
+	 * @param flags If it should be looked in directories or .pak files, or both
+	 * 
+	 * @return If the pattern was valid
+	 */
+	static bool Find(string pattern, inout array<ref CF_File> files, FindFileFlags flags = 2)
+	{
+#ifdef CF_TRACE_ENABLED
+		auto trace = CF_Trace_3("CF_File", "Find").Add(pattern).Add(files).Add(flags);
+#endif
+
+		if (!files)
+		{
+			files = new array<ref CF_File>();
+		}
+
+		string fileName;
+		FileAttr fileAttr;
+		FindFileHandle handle = FindFile(pattern, fileName, fileAttr, flags);
+		if (!handle)
+		{
+			return false;
+		}
+
+		CF_File file;
+
+		// Scuffed do while
+		bool more = true;
+		while (more)
+		{
+			file = new CF_File();
+			file.m_FileName = fileName;
+			file.m_IsDirectory = fileAttr & FileAttr.DIRECTORY;
+			file.m_IsHidden = fileAttr & FileAttr.HIDDEN;
+			file.m_IsReadOnly = fileAttr & FileAttr.READONLY;
+			file.m_IsValid = !(fileAttr & FileAttr.INVALID);
+			files.Insert(file);
+
+			more = FindNextFile(handle, fileName, fileAttr);
+		}
```

Comment:
Turn it into a while(true)

`if(!FindNextFile(handle, fileName, fileAttr)) break;` on L50